pkg/storage/filepath/jsonfile_rest.go

```diff
@@ -244,12 +244,7 @@ func (f *filepathREST) Update(
 		return nil, false, err
 	}
 	if oldVersion != updatedVersion {
```

Comment:
is this check still necessary? seems like you can now fall through to the Write(), and it will handle the conflict correctly.

Reply:
Still needed - it's technically duplicated on K8s side as well (once in `Update()` and once _inside_ etcd 😛 )

**Case 1 (caught by `filepathREST::Update()`**
1. Storage version is `5`
2. Single writer submits update with `resourceVersion: 4`
3. `filepathREST` retrieves object, sees storage version `5`
4. `filepathREST` compares storage version (5) != request version (4) <-- this check
5. If check ⬆️ didn't happen, `MemoryFS::Write` is called with storage version `5` from retrieved object, `MemoryFS` sees that matches what it has, and stale write goes through

--------

**Case 2 (caught by `MemoryFS::Write()` (handled by etcd in K8s case)**
1. Storage version is `5`
2. Writer A submits update with `resourceVersion: 5`
    * `filepathREST` retrieves object, sees storage version `5`
    * `filepathREST` compares storage version (5) == request version (5)
    * Obj version updated to `123`
3. Writer B submits update with `resourceVersion: 5`
    * `filepathREST` retrieves object, sees storage version `5`
    * `filepathREST` compares storage version (5) == request version (5)
    * Obj version updated to `456`
4. `MemoryFS::Write` is called with storage version `5` for Writer A
   * `MemoryFS` compares storage version `5` with passed in version `5` and does write
   * New canonical version of `123` is stored internally
5. `MemoryFS::Write` is called with storage version `5` for Writer B
   * `MemoryFS` compares storage version `123` with passed in version `5` and fails!
   * If check ⬆️ didn't happen, it would overwrite `123` with `456` so both writers would think they succeeded, one going from `5` -> `123` and the other from `5` -> `456` 🙀 

In this case, both got past the `Update()` check since they both retrieved the same storage version, hence the extra check needed at the storage layer.